sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/internal/directconnectivity/GatewayAddressCache.java

```diff
@@ -258,6 +258,11 @@ private URL getServiceEndpoint() {
             headers.put(HttpConstants.HttpHeaders.FORCE_REFRESH, Boolean.TRUE.toString());
         }
 
+        if(request.forceCollectionRoutingMapRefresh)
+        {
+            headers.put(HttpConstants.HttpHeaders.FORCE_COLLECTION_ROUTING_MAP_REFRESH, Boolean.TRUE.toString());
```

Comment:
TRUE.ToString() materialization everytime.

Reply:
We are using Boolean.TRUE.toString() in above FORCE_REFRESH too , and any way this is not major scenario in fact a rare one in which we can that tostring cost